src/js/covidMap.js

```diff
@@ -441,6 +508,182 @@ fluid.defaults("fluid.covidMap.map", {
+        citiesListVisible: true,
+        citiesListExpanded: true,
+        hospitalPanelVisible: false,
+        hospitalPanelExpanded: true
+    }
+});
+
+// The context awareness handler for the mobile view
+fluid.defaults("fluid.covidMap.mobileViewHandler", {
+    gradeNames: "fluid.viewComponent",
+    selectors: {
+        filterCountOnMobile: ".fl-mapviz-filter-count-on-mobile",
+        locationButtonOnMobile: ".fl-mapviz-locations-button-on-mobile",
+        filterButtonOnMobile: ".fl-mapviz-filter-button-on-mobile"
+    },
+    components: {
+        filterCountOnMobile: {
+            type: "fluid.covidMap.filterCount",
+            container: "{that}.dom.filterCountOnMobile"
+        },
+        locationButtonOnMobile: {
+            type: "fluid.button",
+            container: "{that}.dom.locationButtonOnMobile",
+            options: {
+                modelListeners: {
+                    "showCitiesList": {
+                        path: "activate",
+                        changePath: "{map}.model.citiesListVisible",
+                        value: true,
+                        excludeSource: ["init"]
+                    },
+                    "expandCitiesList": {
+                        path: "activate",
+                        changePath: "{map}.model.citiesListExpanded",
+                        value: true,
+                        excludeSource: ["init"]
+                    }
+                }
+            }
+        },
+        filterButtonOnMobile: {
+            type: "fluid.button",
+            container: "{that}.dom.filterButtonOnMobile",
+            options: {
+                modelListeners: {
+                    "showFilterPanel": {
+                        path: "activate",
+                        changePath: "{map}.model.filterPanelVisible",
+                        value: true,
+                        excludeSource: ["init"]
+                    },
+                    "expandFilterPanel": {
+                        path: "activate",
+                        changePath: "{map}.model.filterPanelExpanded",
+                        value: true,
+                        excludeSource: ["init"]
+                    }
+                }
+            }
+        }
+    },
+    model: {
+        filterPanelVisible: false,
+        filterPanelExpanded: true,
+        // resultsPageVisible: false,
+        resultsPageExpanded: true,
+        // citiesListVisible: true,
+        citiesListExpanded: false,
+        hospitalPanelVisible: false,
+        hospitalPanelExpanded: true
+    },
+    modelRelay: {
+        filterPanelVisibilityEffect: {
+            target: "{that}.model",
+            singleTransform: {
+                type: "fluid.transforms.valueMapper",
+                defaultInput: "{that}.model.filterPanelVisible",
+                match: {
+                    true: {
+                        outputValue: {
+                            filterPanelExpanded: true,
```

Comment:
I'm struggling to get hold of the model behind the pattern of flags - it seems like the rule is something like "make the just selected panel visible whilst hiding the others" and I think this could better be expressed by transmitting the relations through a String encoding "visiblePanel": "filter"/"results"/"cities" etc and then separately running a tiny literal modelRelay rule outputting a boolean like so
````
multiplexFilters: {
    source: "visiblePanel",
    func: panel => panel === "filters",
    target: "{map}.model.dom.filterPanel.visible"
}
````
Note that now the visibility/expandedness of the panel is already directly materialised in the model you don't need to allocate separate flags such as "filterPanelVisible" etc. to name them. Also note that we are moving towards the model of "tiny literal functions" rather than the old-fashioned bulky JSON blocks - although the valueMapper is one of the transforms that will probably end up surviving in the the new world, but for different/more complex use cases than this one